cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedInitialization.qll

```diff
@@ -699,11 +699,11 @@ abstract class TranslatedElementInitialization extends TranslatedElement {
   abstract int getElementIndex();
 
   final InstructionTag getElementAddressTag() {
-    result = InitializerElementAddressTag(getElementIndex())
+    result = InitializerElementAddressTag()
```

Comment:
The pair of (ast, instructionTag) has to be unique for each `Instruction`. I believe that with your change, all of the instructions to compute the element addresses for the elements of an initialized array will have both the same AST (the `ArrayAggregateLiteral`) and the same tag (`InitializerElementAddressTag()`).

Reply:
I can see that property was needed before I changed the definition of `TInstruction` in https://github.com/Semmle/ql/pull/866/commits/8ae3551ec18cbd83d9405c09981dcc6f9674653a. Is it needed now?

Reply:
Ah, _that's_ why this didn't break everything! This means there are probably more unnecessary instruction tags sitting around, but that's for another day.